internal/claude/parser.go

```
@@ -227,46 +228,59 @@ func getStringValue(data map[string]interface{}, key string) string {
 }
 
-// groupEventsByProject groups events by project directory
+// groupEventsByProject groups events by project directory or git repository based on threads flag
 func groupEventsByProject(events []*models.SessionEvent, threads bool) ([]*models.SessionTimeline, error) {
-	// This is a simplified version - we'll implement the full grouping logic later
-	// For now, just group by directory
-	projectMap := make(map[string][]*models.SessionEvent)
+	if threads {
+		// threads=true (worktree mode): group by git repository with child project support
+		return groupEventsByRepositoryWithChildren(events)
+	} else {
```

golangci-lint (1.64.8)

[warning] indent-error-flow: if block ends with a return statement, so drop this else and outdent its block

(revive)